src/components/ReportingConfig/ReportingConfigForm.jsx

```diff
@@ -157,148 +232,95 @@ class ReportingConfigForm extends React.Component {
+              helpText: 'The frequency interval (daily, weekly, or monthly) that the report should be sent',
+              label: 'Frequency',
+              options: reportingConfigTypes.frequency.map(item => ({ label: item[1], value: item[0] })),
+              defaultValue: frequency,
+              onChange: event => this.setState({ frequency: event.target.value }),
+            })}
           </div>
         </div>
         <div className="row">
           <div className="col">
-            <ValidationFormGroup
-              for="dayOfMonth"
-              helpText="The day of the month to send the report. This field is required and only valid when the frequency is monthly"
-              invalid={frequency === 'monthly' && invalidFields.dayOfMonth}
-            >
-              <label htmlFor="dayOfMonth">Day of Month</label>
-              <Input
-                type="number"
-                max={MONTHLY_MAX}
-                min={MONTHLY_MIN}
-                disabled={!(frequency === 'monthly')}
-                id="dayOfMonth"
-                name="dayOfMonth"
-                defaultValue={config ? config.dayOfMonth : 1}
-                onBlur={e => this.handleBlur(e)}
-              />
-            </ValidationFormGroup>
+            {this.renderNumberField({
+              key: 'dayOfMonth',
+              helpText: 'The day of the month to send the report. This field is required and only valid when the frequency is monthly',
+              isInvalid: frequency === 'monthly' && invalidFields.dayOfMonth,
+              label: 'Day of Month',
+              max: MONTHLY_MAX,
+              min: MONTHLY_MIN,
+              disabled: !(frequency === 'monthly'),
+              onBlur: event => this.handleBlur(event),
+            })}
           </div>
           <div className="col">
-            <ValidationFormGroup
-              for="dayOfWeek"
-              helpText="The day of the week to send the report. This field is required and only valid when the frequency is weekly"
-            >
-              <label htmlFor="dayOfWeek">Day of Week</label>
-              <Input
-                type="select"
-                id="dayOfWeek"
-                name="dayOfWeek"
-                disabled={!(frequency === 'weekly')}
-                options={reportingConfigTypes.dayOfWeek.map(item => ({ label: item[1], value: item[0] }))}
-                defaultValue={config ? config.dayOfWeek : undefined}
-              />
-            </ValidationFormGroup>
+            {this.renderSelect({
+              key: 'dayOfWeek',
+              helpText: 'The day of the week to send the report. This field is required and only valid when the frequency is weekly',
+              label: 'Day of Week',
+              options: reportingConfigTypes.dayOfWeek.map(item => ({ label: item[1], value: item[0] })),
+              disabled: !(frequency === 'weekly'),
+              // onChange: event => this.setState({ deliveryMethod: event.target.value }),
```

Comment:
Is this intentionally commented out?

Reply:
no it was copied from another place, that's why this comment left. Removing now